### DISCUSSION TITLE
如何让服务端命令走搭建好的代理

### DISCUSSION THREAD

**Discussion by dzhl** [v2fly/v2ray-core#2505] [Q&A]
这是一个场景描述：在VPS A上搭建了V2RAY服务，通过配置文件中的“套娃”实现对奈非进行分流。对于奈非的流量会通过“套娃”走到VPS B。我想要在VPS A上运行Netflix解锁程序（https://github.com/sjlleo/netflix-verify），以判断是否仍然可以解锁Netflix。但如果直接在VPS A上运行netflix-verify，则仍然会使用VPS A进行奈非解锁判断。有没有办法让在VPS A上运行netflix-verify时，也能够像v2ray客户端一样使用由VPS A搭建的代理服务呢？我的表述可能比较混乱，请问大家是否明白我的意思？

**Comment by mydogshitgold** [comment]
Yes, the main part of the description is quite erratic. Can you rewrite it to be clearer and break it into **smaller parts** with more contextual points?

**Reply by dzhl** [reply]
> Yes, the main part of the description is quite erratic. Can you rewrite it to be clearer and break it into **smaller parts** with more contextual points?

This is a scene description: V2RAY service is set up on VPS A, and "nested" routing is implemented through the configuration file to divert traffic from Netflix. The traffic for Netflix will be routed to VPS B via "nesting". I want to run the Netflix unlock program (https://github.com/sjlleo/netflix-verify) on VPS A to determine if it can still unlock Netflix. However, if netflix-verify is directly run on VPS A, it will still use VPS A for unlocking Netflix. Is there any way to make netflix-verify running on VPS A also use the proxy service built by VPS A like v2ray client? My expression may be confusing, do you understand what I mean?

**Reply by mydogshitgold** [reply]
So you are using VPS A as a relay for VPS B for all inbound Netflix traffic, and you want to run a test program on VPS A to verify that VPS B still has access to Netflix? Correct?

**Reply by dzhl** [reply]
> So you are using VPS A as a relay for VPS B for all inbound Netflix traffic, and you want to run a test program on VPS A to verify that VPS B still has access to Netflix? Correct?

yes,it's correct

**Reply by mydogshitgold** [reply]
It appears that the tool you are using [supports http proxy](https://github.com/sjlleo/netflix-verify/pull/11/files). You can simply open an http inbound on **VPS A** and route it to **VPS B** outbound, then run the said tool with `env http_proxy=http://127.0.0.1:your_port https_proxy=http://127.0.0.1:your_port ./netflix-verify`

**Reply by mydogshitgold** [reply]
Here is a sample configuration snippet you can use as a reference:
```json
{
  "inbounds": [
    {
      "tag": "vps_b_inbound",
      "protocol": "http",
      "listen": "127.0.0.1",
      "port": 30001
    }
  ],
  "outbounds": [
    {
      "tag": "vps_b_outbound"
    }
  ],
  "routing": {
    "rules": [
      {
        "outboundTag": "vps_b_outbound",
        "inboundTag": [
          "vps_b"
        ],
        "type": "field"
      }
    ]
  }
}
```

**Reply by dzhl** [reply]
> env http_proxy=http://127.0.0.1:your_port

it not worked

**Reply by mydogshitgold** [reply]
Can you tell me what command you are using and what the output is? 
What is your current configuration? 

---

The placeholder `your_port` in the command `env http_proxy=http://127.0.0.1:your_port` is intended to be substituted with the port number that you have configured for the HTTP proxy.

**Comment by dzhl** [comment]
command is from https://github.com/sjlleo/netflix-verify/releases/download/v3.1.0-1/nf_linux_amd64
Inputbound：
```
"inbounds": [
        {
            "port": 10241, 
            "protocol": "vless", 
            "tag": "VLESSTCP", 
            "settings": {
                "clients": [
                    {
                        "id": "xxxxxxxxxxx", 
                        "add": "11111", 
                        "email": "xxxxxxxxxx"
                    }
                ], 
                "decryption": "none", 
                "fallbacks": [
                    {
                        "dest": 31296, 
                        "xver": 1
                    }, 
                    {
                        "alpn": "h2", 
                        "dest": 31302, 
                        "xver": 0
                    }
                ]
            }, 
            "streamSettings": {
                "network": "tcp", 
                "security": "tls", 
                "tlsSettings": {
                    "minVersion": "1.2", 
                    "alpn": [
                        "http/1.1", 
                        "h2"
                    ], 
                    "certificates": [
                        {
                            "certificateFile": "/etc/v2ray-agent/tls/xxxxx.crt", 
                            "keyFile": "/etc/v2ray-agent/tls/xxx.key", 
                            "ocspStapling": 3600, 
                            "usage": "encipherment"
                        }
                    ]
                }
            }, 
            "sniffing": {
                "enabled": true, 
                "destOverride": [
                    "http", 
                    "tls"
                ]
            }
        },
        {
      "tag": "streamingMedia-443",
      "protocol": "http",
      "listen": "127.0.0.1",
      "port": 30001
    }

    ]
```
outbound：
```
  "outbounds":[
        {
            "protocol":"freedom",
            "settings":{
                "domainStrategy":"UseIPv4"
            },
            "tag":"IPv4-out"
        },
        {
            "protocol":"freedom",
            "settings":{
                "domainStrategy":"UseIPv6"
            },
            "tag":"IPv6-out"
        },
        {
            "protocol":"blackhole",
            "tag":"blackhole-out"
        }, 
         // 这是通过任意门解锁
        {
            "tag": "streamingMedia-80", 
            "protocol": "freedom", 
            "settings": {
                //如果不是IPV6解锁，值改为UseIPv4
                "domainStrategy": "UseIPv4", 
                
                "redirect": "VPS B"
            }
        }, 
        {
            "tag": "streamingMedia-443", 
            "protocol": "freedom", 
            "settings": {
                //如果不是IPV6解锁，值改为UseIPv4
                "domainStrategy": "UseIPv4", 
                "redirect": "VPS B"
            }
        }
    ]
```
route：
```
"routing": {
        "domainStrategy": "AsIs", 
        "rules": [
             
            #下面是通过任意门解锁，和上面解锁方式只保留一个
            {
                "type": "field", 
                "port": 80, 
                "domain": [
                    "geosite:netflix",
                    "geosite:disney", 
                    "ip.sb"
                ], 
                "outboundTag": "streamingMedia-80"
            }, 
            {
                "type": "field", 
                "port": 80, 
                "ip": [
                    "geoip:netflix",
                    "249.249.249.249"
                ], 
                "outboundTag": "streamingMedia-80"
            },
            {
                "type": "field", 
                "port": 443, 
                "domain": [
                    "geosite:netflix", 
                    "geosite:disney",
                    "ip.sb"
                ], 
                "outboundTag": "streamingMedia-443"
            },
            {
                "type": "field", 
                "port": 443, 
                "ip": [
                    "geoip:netflix",
                    "249.249.249.249"
                ],
                "outboundTag": "streamingMedia-443"
            }
        ]
    }
```

**Reply by mydogshitgold** [reply]
Can you reformat your post, it's hard to read? Single back quote does not work for multi-line markdown code block.

> command is from https://github.com/sjlleo/netflix-verify/releases/download/v3.1.0-1/nf_linux_amd64

By command, I mean the shell command you are using, not the program.

**Reply by mydogshitgold** [reply]
@nursery01 Might I implore your pardon, my liege, to suspend from sending such strikingly spooky statements succeeding my sayings? Your mercy in mitigating this matter would be most magnificently memorable.

**Comment by dzhl** [comment]
why i can not see reply in this page？
